### ISSUE TITLE
Sampler management in alodataset train_loader not generic

### ISSUE THREAD

**Issue by ragier** [Visual-Behavior/aloception#107]
Inside train_loader, sampler arg is called with `dataset` as arguments, but other samplers need other arguments (like SubsetRandomSampler)

The sampler should be constructed before the call, but we don't not have the dataset yet (for the random sampler for instance).

**Comment by thibo73800** [comment]
@jsalotti  @anhtu293  do you think this is still relevant or not ? If not, please, close it. Otherwise, please, add labels on this issue

**Comment by jsalotti** [comment]
I do not have in mind a case for which we currently need `SubsetRandomSampler` or another sampler with kwargs.

- One solution could be to add a `sampler_kwargs` to `train_loader`.
- But without changing the current code, one could call `train_loader` and give a function as  `sampler` parameter, that will construct the sampler based on `dataset` if needed.

**Comment by anhtu293** [comment]
@jsalotti can i close this issue ?
